lib/NetworkClient.js

```diff
@@ -19,7 +19,11 @@ function readRequest(requestData)
             return;
         }
         if (!error && response.statusCode === 200) {
-            d.resolve(body);
+            if (!requestData.method || requestData.method != 'HEAD') {
+                d.resolve(body);
```

Comment:
The returned promise value now contains: {body: body, headers: response.headers}.